rust/private/rust.bzl

```diff
@@ -543,7 +543,10 @@ _common_attrs = {
         """),
     ),
     "rustc_flags": attr.string_list(
-        doc = "List of compiler flags passed to `rustc`.",
+        doc = _tidy("""
+            List of compiler flags passed to `rustc` for this target. For flags that should be
```

Comment:
Could you mention that flags specified in the attribute are not applied to dependencies of the target? Folks often get this wrong.

I wonder if we should mention codegen flags in rustc_flags attribute. Ideally we would have full symmetry and full coverage, for example:

* extra_rustc_flags attribute and @rules_rust//:extra_rustc_flags setting
* extra_codegen_flags attribute and @rules_rust//:extra_codegen_flags setting

And then we would only mention the setting in the relevant attribute.

(but also see my bikeshed comment about naming above :)

Reply:
Ah, because rustc_flags is more general than codegen, specifically -- that makes sense. Do you think instead of an extra codegen flag this should just be extra rustc flags, then?

Reply:
Yeah I can imagine having both at some point, but starting with `extra_rustc_flags` makes sense to me. Or `rustcopts` :)